### ISSUE TITLE
Question: Flow-DPM-Solver

### ISSUE THREAD

**Issue by dxqbYD** [NVlabs/Sana#110]
Is there already a generic implementation of your Flow-DPM-Solver, such as for Comfy or Swarm?
Would you expect it to work equally well for other flow matching models such as Flux?

thank you

edit: I might have misunderstood your paper. on second read, it sounds like you have adapted DPM to flow matching and compared it to Euler - not proposed a new inference method.

**Comment by lawrence-cj** [comment]
Of course. We just adapt DPM to flow matching scheduler and it can works for FLUX with changes. Now the Flow-DPM-Solver is merged into diffusers pipeline.
refer to: https://github.com/huggingface/diffusers/blob/3c2e2aa8a902ebaf57ea36e48a64b52dc9b2e7df/scripts/convert_sana_to_diffusers.py#L232

**Comment by dxqbYD** [comment]
thanks for your answer!

I quickly tried it on Flux without changes, and subjectively it does seem to have a lot more sharp details already at around 18 steps compared to Euler.
On the other hand, it seems you cannot really use it for going below 20 steps. For example, DPM at 14 steps isn't close to Euler at 18 but it's still a completely different image.

**Comment by lawrence-cj** [comment]
DPM <15 steps will degrade to DDIM. that’s why it doesn’t work. Could you share some comparison results here between Flow DPMS and Euler on FLUX? Curious about it.

**Comment by dxqbYD** [comment]
sure
First FlowMatchEulerDiscreteScheduler initialized from Flux's scheduler/.json, second yours initialized from Sana's scheduler/.json
Both 18 steps

![image](https://github.com/user-attachments/assets/e63a28fb-3848-43df-b8b3-1920af7d97dc)
![image](https://github.com/user-attachments/assets/901d04f7-26ee-40a2-936f-524628329ff0)

**Comment by lawrence-cj** [comment]
I don’t see any difference😂

**Comment by dxqbYD** [comment]
> I don’t see any difference😂

it's subtle, open in separate tabs and switch
was the difference higher in your Sana tests?
